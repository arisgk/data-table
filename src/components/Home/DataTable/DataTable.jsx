import React from 'react';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';

const DataTable = () => (
  <Table>
    <TableHeader>
      <TableRow>
        <TableHeaderColumn>ID</TableHeaderColumn>
        <TableHeaderColumn>First Name</TableHeaderColumn>
        <TableHeaderColumn>Last Name</TableHeaderColumn>
        <TableHeaderColumn>Age</TableHeaderColumn>
      </TableRow>
    </TableHeader>
    <TableBody>
      <TableRow>
        <TableRowColumn>1</TableRowColumn>
        <TableRowColumn>John</TableRowColumn>
        <TableRowColumn>Smith</TableRowColumn>
        <TableRowColumn>36</TableRowColumn>
      </TableRow>
    </TableBody>
  </Table>
);

export default DataTable;
