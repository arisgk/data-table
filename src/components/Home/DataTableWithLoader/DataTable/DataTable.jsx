import React from 'react';
import PropTypes from 'prop-types';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';
import * as schemas from '../../../../schemas/propTypes';

const DataTable = ({ users }) => (
  <Table>
    <TableHeader>
      <TableRow>
        <TableHeaderColumn>ID</TableHeaderColumn>
        <TableHeaderColumn>First Name</TableHeaderColumn>
        <TableHeaderColumn>Last Name</TableHeaderColumn>
        <TableHeaderColumn>Age</TableHeaderColumn>
      </TableRow>
    </TableHeader>
    <TableBody>
      {
        users
          ? users.map(user => (
            <TableRow key={user.id}>
              <TableRowColumn>{user.id}</TableRowColumn>
              <TableRowColumn>{user.firstName}</TableRowColumn>
              <TableRowColumn>{user.lastName}</TableRowColumn>
              <TableRowColumn>{user.age}</TableRowColumn>
            </TableRow>
          ))
          : null
      }
    </TableBody>
  </Table>
);

DataTable.propTypes = {
  users: PropTypes.arrayOf(schemas.user),
};

DataTable.defaultProps = {
  users: [],
};

export default DataTable;
