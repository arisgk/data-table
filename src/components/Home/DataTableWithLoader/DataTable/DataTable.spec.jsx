/* global describe, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DataTable from './DataTable';

describe('DataTable', () => {
  it('renders correctly', () => {
    const users = [
      {
        id: '1234abcd',
        firstName: 'Lebron',
        lastName: 'James',
        age: 32,
      },
      {
        id: '2345bcde',
        firstName: 'Steph',
        lastName: 'Curry',
        age: 29,
      },
    ];

    const wrapper = shallow(
      <DataTable users={users} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
