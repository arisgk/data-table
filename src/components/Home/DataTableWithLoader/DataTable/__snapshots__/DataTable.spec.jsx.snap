// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable renders correctly 1`] = `
<Table
  allRowsSelected={false}
  fixedFooter={true}
  fixedHeader={true}
  height="inherit"
  multiSelectable={false}
  selectable={true}
>
  <TableHeader
    adjustForCheckbox={true}
    displaySelectAll={true}
    enableSelectAll={true}
    selectAllSelected={false}
  >
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      selectable={true}
      striped={false}
    >
      <TableHeaderColumn>
        ID
      </TableHeaderColumn>
      <TableHeaderColumn>
        First Name
      </TableHeaderColumn>
      <TableHeaderColumn>
        Last Name
      </TableHeaderColumn>
      <TableHeaderColumn>
        Age
      </TableHeaderColumn>
    </TableRow>
  </TableHeader>
  <TableBody
    allRowsSelected={false}
    deselectOnClickaway={true}
    displayRowCheckbox={true}
    multiSelectable={false}
    preScanRows={true}
    selectable={true}
    style={Object {}}
  >
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        hoverable={false}
      >
        1234abcd
      </TableRowColumn>
      <TableRowColumn
        hoverable={false}
      >
        Lebron
      </TableRowColumn>
      <TableRowColumn
        hoverable={false}
      >
        James
      </TableRowColumn>
      <TableRowColumn
        hoverable={false}
      >
        32
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        hoverable={false}
      >
        2345bcde
      </TableRowColumn>
      <TableRowColumn
        hoverable={false}
      >
        Steph
      </TableRowColumn>
      <TableRowColumn
        hoverable={false}
      >
        Curry
      </TableRowColumn>
      <TableRowColumn
        hoverable={false}
      >
        29
      </TableRowColumn>
    </TableRow>
  </TableBody>
</Table>
`;
